deliver/lib/deliver/upload_assets.rb

```diff
@@ -6,17 +6,28 @@ def upload(options)
       v = app.edit_version
       UI.user_error!("Could not find a version to edit for app '#{app.name}'") unless v
 
+      # check if we need to call save! and waste time
+      save_needed = false
+
       if options[:app_icon]
-        UI.message("Uploading app icon...")
-        v.upload_large_icon!(options[:app_icon])
+        if v.upload_large_icon!(options[:app_icon])
```

Comment:
`upload_large_icon` already uploads the icon, right? Which means we already have a waiting time here, but don't properly show the `UI.message` (since it's being called afterwards). 
Not a big deal, I guess we could just leave it like this, since we shouldn't print anything from `spaceship`